soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -61,21 +61,22 @@ static ALWAYS_INLINE void clock_init(void)
 	/* Enables the clock for the I/O controller.: Enable Clock. */
     CLOCK_EnableClock(kCLOCK_Iocon);
 
-#ifdef CONFIG_I2C_4
+#if DT_HAS_NODE(DT_NODELABEL(flexcomm4)) && \
+    DT_NODE_HAS_COMPAT(DT_NODELABEL(flexcomm4), nxp_lpc_i2c)
 	/* attach 12 MHz clock to FLEXCOMM4 */
 	CLOCK_AttachClk(kFRO12M_to_FLEXCOMM4);
 
 	/* reset FLEXCOMM for I2C */
 	RESET_PeripheralReset(kFC4_RST_SHIFT_RSTn);
-#endif /* CONFIG_I2C_4 */
+#endif
 
-#ifdef CONFIG_SPI_8
+#if DT_HAS_NODE(DT_NODELABEL(hs_lspi))
 	/* Attach 12 MHz clock to FLEXCOMM8 */
```

Comment:
Should we change the comment here as well to not refer to `FLEXCOMM8`?

Reply:
Done.